oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/index/ElasticsearchIndexHelper.java

```diff
@@ -0,0 +1,141 @@
+import org.apache.jackrabbit.oak.plugins.index.elasticsearch.ElasticsearchIndexDefinition;
+import org.apache.jackrabbit.oak.plugins.index.search.FieldNames;
+import org.apache.jackrabbit.oak.plugins.index.search.PropertyDefinition;
+import org.elasticsearch.client.indices.CreateIndexRequest;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Provides utility functions around Elasticsearch indexing
+ */
+class ElasticsearchIndexHelper {
+
+    public static CreateIndexRequest createIndexRequest(ElasticsearchIndexDefinition indexDefinition) throws IOException {
+        final CreateIndexRequest request = new CreateIndexRequest(indexDefinition.getRemoteIndexName());
+
+        // provision settings
+        // https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-pathhierarchy-tokenizer.html
+        request.settings(Settings.builder()
+                .put("analysis.analyzer.ancestor_analyzer.type", "custom")
+                .put("analysis.analyzer.ancestor_analyzer.tokenizer", "path_hierarchy"));
+
+        // provision mappings
+        final XContentBuilder mappingBuilder = XContentFactory.jsonBuilder();
+        mappingBuilder.startObject();
+        {
+            mappingBuilder.startObject("properties");
+            {
+                mapInternalProperties(mappingBuilder);
+                mapIndexRules(indexDefinition, mappingBuilder);
+            }
+            mappingBuilder.endObject();
+        }
+        mappingBuilder.endObject();
+        request.mapping(mappingBuilder);
+
+        return request;
+    }
+
+    private static void mapInternalProperties(XContentBuilder mappingBuilder) throws IOException {
+        mappingBuilder.startObject(FieldNames.PATH)
+                .field("type", "keyword")
+                .endObject();
+        mappingBuilder.startObject(FieldNames.ANCESTORS)
+                .field("type", "text")
+                .field("analyzer", "ancestor_analyzer")
+                .field("search_analyzer", "keyword")
+                .field("search_quote_analyzer", "keyword")
+                .endObject();
+        mappingBuilder.startObject(FieldNames.PATH_DEPTH)
+                .field("type", "integer")
+                .endObject();
+//                mappingBuilder.startObject(FieldNames.SUGGEST)
+//                        .field("type", "completion")
+//                        .endObject();
+//                mappingBuilder.startObject(FieldNames.NOT_NULL_PROPS)
+//                        .field("type", "keyword")
+//                        .endObject();
+//                mappingBuilder.startObject(FieldNames.NULL_PROPS)
+//                        .field("type", "keyword")
+//                        .endObject();
```

Comment:
mappingBuilder properties commented by mistake?

Reply:
I have actually done it on purpose. Those mappings are related to some features we don't currently support at query time (eg: autosuggest). Since I am not sure the mapping is correct I have preferred to comment it for now. They need to be reviewed when the specific features will be implemented. I have added a TODO comment to clarify it.